### ISSUE TITLE
[ReadMe] RyuJIT = Microsoft.NETCore.Jit = ?

### ISSUE THREAD

**Issue by ShalokShalom** [dotnet/coreclr#14563]
Is the mentioned Microsoft.NETCore.Jit RyuJIT?

If so, name it so. 

Otherwise, the already confusing situation with a lot of different compilers gets one step more confusing. 

Thanks a lot

**Comment by ShalokShalom** [comment]
And where is the actual source code?

All I can find is the nuget package, Microsoft.NETCore.Jit in "Find file" here at Github brings us nothing else.

**Comment by mikedn** [comment]
> If so, name it so.

Name what so? Microsoft.NETCore.Jit is the NuGet package name, it's unlikely that someone will go and change that now. RyuJIT is the codename of the current .NET Core (and partially .NET Framework too) JIT. The 2 names represent different things.

**Comment by jkotas** [comment]
> Microsoft.NETCore.Jit is the NuGet package name, it's unlikely that someone will go and change that now.

Also, we prefer to not use codenames in the package names.

**Comment by ShalokShalom** [comment]
So, where can I find RyuJIT? And why is this the name which is used everywhere - except in the actual source code?

**Comment by ShalokShalom** [comment]
And why this: https://github.com/dotnet/corefx#net-core-libraries-corefx

The corresponding .NET Core Runtime repo (called "CoreCLR") contains the runtime implementation for .NET Core. It includes RyuJIT, the .NET GC, and many other components. 

And then no mention about it here in this repo?

**Comment by jkotas** [comment]
https://github.com/dotnet/coreclr/tree/master/src/jit

**Comment by ShalokShalom** [comment]
> RyuJIT is the codename of the current .NET Core (and partially .NET Framework too) JIT. The 2 names represent different things.

So, what is the package name of RyuJIT?

**Comment by mikedn** [comment]
> So, what is the package name of RyuJIT?

Microsoft.NETCore.Jit.

It's not very irrelevant. Much like the name of the class that implements an interface is not relevant to the consumers of the interface.

There's also the little detail that the current ARM32 JIT is a sort of Frankenstein that uses the RyuJIT optimization phases and the legacy JIT32 register allocator and codegen.

And the little detail that .NET Framework uses RyuJIT only on x64 while on x86 it uses the legacy JIT32. And on x64 it can also use the legacy JIT64. And there's no `Microsoft.NETCore.Jit` relevant to the .NET Framework. Not any other NuGet package.

So for everyone who wants to follow the current .NET JIT development/progress/whatever the relevant name is RyuJIT. Microsoft.NETCore.Jit is just an abstraction.

**Comment by ShalokShalom** [comment]
So, you write its something different and then that it is the same?

This whole thing feels like somebody is sitting in a room for hours, figuring out how to name it so complicate as possible.

**Comment by jkotas** [comment]
You are right that people often sit in a ~~room~~ pub for hours to come up with a good [codename]( https://en.wikipedia.org/wiki/Code_name).
